log4j-core/src/main/java/org/apache/logging/log4j/core/config/DefaultConfiguration.java

```diff
@@ -49,4 +61,60 @@ public DefaultConfiguration() {
 
     @Override
     protected void doConfigure() {}
+
+    static Layout<? extends String> createDefaultLayout() {
+        return new DefaultLayout();
+    }
+
+    /**
+     * A simple layout used only by {@link DefaultConfiguration}
+     * <p>
+     *   This layout allows to create applications that don't contain {@link org.apache.logging.log4j.core.layout.PatternLayout}
+     *   and all its patterns, e.g. GraalVM applications.
+     * </p>
+     */
+    private static final class DefaultLayout extends AbstractLayout<String> {
+
+        private static final byte[] EMPTY_BYTE_ARRAY = new byte[0];
```

Comment:
Do we really need this optimization? Can't we simply inline `new byte[0]` to call sites?

Reply:
We can do better and use `null`.

Fixed in https://github.com/apache/logging-log4j2/pull/3118/commits/bf892e169e273d3e9c4e4812e4ffac4178460a3d